### PR TITLE
[Tests] Snackbar Unit Tests

### PR DESCRIPTION
**Pull request by TimRoe**
## Description of Change

Created a suite of ~~32~~ 34 unit tests for both Snackbar UI itself (21 tests) and the SnackbarProvider wrapper (~~11~~ 13 tests) to validate functionality.

~~One edge case unit tests were not cooperating with is validating the standard workflow around the Snackbar persisting. While it works fine with the screen reader enabled to persist for 5 seconds, with the screen reader disabled or screen reader enabled w/ action button where it should persist essentially forever (1 billion seconds) it was disappearing, for some reason, very quickly (less than a second). A separate ticket will be created to explore this test case further. It's believed to be related to using fake timers even though they should expedite time 1:1 (AKA it should persist far longer than the desired test case of 6 seconds to validate it persisted longer than the 5 seconds).~~
^^ Sorted out... tl;dr: Jest runs the unit tests in node which our 1 billion seconds "indefinite" was overflowing, adjusting to node's 32-bit integer max (-1) of 2147483646 works and is still 23ish days so should be fine for practical purposes.

## Testing

Spot checked iOS/Android/Web. Validated iOS/Android work to have 0 for custom offset and "Undo"/"Try again" display as expected with i18n wrapper.

Working through unit tests, tested values that ought to cause every test to fail and failures occurred as expected.

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [x] Merge `main` into branch
- [x] Merge branch to `main`
- [x] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [x] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

File: packages/components/src/components/Snackbar/useSnackbar.tsx [hunk 1]
```diff
@@ -25,8 +25,10 @@ export function useSnackbar() {
   const show = (message: string, snackbarOptions?: SnackbarOptions) => {
     const { offset, setOffset } = context
 
-    // Adjust offset if provided or reset to default
-    const newOffset = snackbarOptions?.offset || defaultOffset
+    const customOffset = snackbarOptions?.offset
+
+    // Custom offset if provided, else default
+    const newOffset = customOffset === undefined ? defaultOffset : customOffset
```
Comment (TimRoe) [inline packages/components/src/components/Snackbar/useSnackbar.tsx:31]: Fixing bug with offset `0` setting the offset to 60 (defaultOffset) due to 0 being falsy.

File: packages/components/src/components/Snackbar/Snackbar.tsx [hunk 2]
```diff
@@ -238,18 +239,20 @@ export const Snackbar: FC<SnackbarProps> = (toast) => {
   }
 
   return (
-    <View {...containerProps}>
-      <View {...iconAndMessageContainer}>
-        <View style={iconViewStyle}>
-          <Icon {...iconProps} />
+    <ComponentWrapper>
```
Comment (TimRoe) [inline packages/components/src/components/Snackbar/Snackbar.tsx:242]: Just adding ComponentWrapper to ensure component always is pulling i18n similar to other components. Worked fine in-app because they already had the requisite translations.

File: packages/components/src/components/Snackbar/SnackbarProvider.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ type SnackbarContextType = {
 }
 
 export const SNACKBAR_DURATIONS = {
-  DEFAULT: 1000000000000, // Essentially indefinite until dismissed
+  DEFAULT: 2147483646, // Essentially indefinite until dismissed (23 days, limited by 32-bit int)
```
Comment (TimRoe) [inline packages/components/src/components/Snackbar/SnackbarProvider.tsx:15]: Not an issue with the code itself, but the unit tests run in node which only allows 32-bit integers which go from -2147483647 to 2147483647 and the overflow was setting it to 1 second. Adjusted down to make the unit tests work and 23 days and change is still plenty to be "indefinite" for practical purposes.

**Review by narin — APPROVED** [review]
Looks good! Very thorough.
